src/components/SubPage/convert-data.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    Diagram,
+    Clone,
+    DiseaseCellLineNode,
+} from "../../component-queries/types";
+import { DiagramCardProps } from "../shared/DiagramCard";
+import {
+    ClonePercentPositive,
+    UnpackedDiseaseCellLineFull,
+    UnpackedEditingDesign,
+} from "./types";
+
+export const unpackDiagrams = (diagrams?: Diagram[]): DiagramCardProps[] => {
+    if (!diagrams || diagrams.length === 0) {
+        return [];
+    }
+    return diagrams.map((diagram) => {
+        return {
+            title: diagram.title,
+            caption: diagram.caption,
+            image: diagram.image.childImageSharp.gatsbyImageData,
+        };
+    });
+};
+
+// subpage data
+export const unpackEditingDesignData = (editing_design?: {
+    crna_target_site?: string;
+    dna_donor_sequence?: string;
+    cas9?: string;
+    f_primer?: string;
+    r_primer?: string;
+    diagrams?: Diagram[];
+}): UnpackedEditingDesign => {
+    if (!editing_design) {
+        return {};
+    }
+    return {
+        crnaTargetSite: editing_design.crna_target_site,
+        dnaDonorSequence: editing_design.dna_donor_sequence,
+        cas9: editing_design.cas9,
+        fPrimer: editing_design.f_primer,
+        rPrimer: editing_design.r_primer,
+        diagrams: unpackDiagrams(editing_design.diagrams),
+    };
+};
+
+export const getStemCellCharData = (clones: Clone[]) => {
+    const init: {
+        percentPositive: ClonePercentPositive[];
+        passingAntibodies: any[];
+        differentiation: any[];
+    } = {
+        percentPositive: [],
+        passingAntibodies: [],
+        differentiation: [],
+    };
+    return clones.reduce((acc, clone) => {
+        const cloneNumber = clone.clone_number;
+        if (cloneNumber === undefined) {
+            return acc;
+        }
+        if (clone.positive_cells !== undefined) {
+            acc.percentPositive.push({
+                cloneNumber,
+                value: clone.positive_cells,
+            });
+        }
+        return acc;
+    }, init);
+};
```

Comment:
this is the only new logic, the rest of this document is code taken from other places and consolidated here